app/routes/shared.py

```
@@ -216,8 +219,17 @@ async def execute_query(query: str, params: Optional[List[Any]] = None, fetch_on
                 result = await conn.fetch(query, *(params or []))
             return result
 
-    except Exception as e:
-        logging.error(f"Database operation failed: {e}")
+    except asyncio.TimeoutError:
+        logger.error(
+            "DB pool acquire timed out (pool exhausted) — all 30 connections busy\nCaller stack:\n%s",
+            ''.join(traceback.format_stack()[:-1])
+        )
+        raise HTTPException(status_code=status.HTTP_503_SERVICE_UNAVAILABLE, detail="Database busy")
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)